engineapi/disk_service.go

```diff
@@ -59,22 +59,22 @@ func (s *DiskService) Close() {
 	}
 }
 
-func (s *DiskService) DiskCreate(diskType, diskName, diskUUID, diskPath string, blockSize int64) (*imapi.DiskInfo, error) {
-	return s.grpcClient.DiskCreate(diskType, diskName, diskUUID, diskPath, blockSize)
+func (s *DiskService) DiskCreate(diskType, diskName, diskUUID, diskPath, diskDriver string, blockSize int64) (*imapi.DiskInfo, error) {
+	return s.grpcClient.DiskCreate(diskType, diskName, diskUUID, diskPath, diskDriver, blockSize)
```

Comment:
For the existing block-type disks, we patched the diskDriver in the upgrade path, so they will still use AIO bdev and we don't need to do any change in the IM pod.

Reply:
I mean, the old running IM pods (for v2 engine). Can the new longhorn-manager pods talk with the old IM pods and users be able to move the existing v2 volumes to the old IM pods after API changes?

Reply:
Got your point.
Updated the codes for handling the compatibility of a new longhorn-manager with an old longhorn-instance-manager.

Reply:
Which commit do you add the compatibility code to?

Reply:
Actually, no. I fixed the bugs when testing the compatibility.
`NewDiskInfo(diskName, ....` => The diskName is sometimes empty, so users cannot create and delete volumes when talking to an old IM.

We don't need to update other parts, because the `diskDriver` is not supported in old gRPC server.
Old IM pod processes the disks using the AIO method.